plugins/modules/win_unzip.py

```diff
@@ -49,6 +49,11 @@
     description:
       - If this file or directory exists the specified src will not be extracted.
     type: path
+  password:
+    description:
+      - If a zip file is encrypted with password.
+      - Passing a value to a password parameter requires the PSCX module to be installed.
+      - The I(recurse) option must be C(yes).
```

Comment:
So this is the only part that I'm unsure as to why it is needed? Is `recurse: yes` only required so it goes the PSCX path. If that's the case you should add a conditional to https://github.com/ansible-collections/community.windows/blob/2c6d6dbc6654b4d3634f0f67907d2265147d382b/plugins/modules/win_unzip.ps1#L106 to be

```
If ($ext -eq ".zip" -And $recurse -eq $false -and -not $password) {
```

This means that if the `password` is set then it will never try and extract it using the builtin tools and will immediately use PSCX for that.

Reply:
Yes, I made `recurse: yes` only so it goes to the PSCX path and skips normal build-in tools. As we can see the doc part it clearly mentions that if you set `recurse: yes` then you need PSCX module to be installed on your target machine.  We can add the above condition which runs only when `password` is set.

Reply:
Yep, because the builtin tools with Windows don't seem to support extracting a password protected zip we can make sure we always ensure that when `password` is set it goes the PSCX route removing the need for `recurse: yes` to also be set.